utils.py

```diff
@@ -189,16 +189,63 @@ def validate_cookies_del(flows: list[dict]) -> None:
             )
 
 
-def flows_to_log(logger_fun: Callable, message: str, flow_dict: dict[str, list]):
-    """Log flows, maximun flows in a log is 200"""
-    length_msg = f"total_length: {len(flow_dict['flows'])}, "
+def _flows_to_log(
+    logger_fun: Callable,
+    flows: list,
+    dpid: Union[str, list],
+    counter: int,
+    flows_acc: dict,
+    maximun: int,
+    message: str,
+    total_length: str,
+) -> tuple[int, dict, str]:
+    start, end = 0, counter
+    while flows[start:end]:
+        counter -= len(flows[start:end])
+        flows_acc[str(dpid)] = flows[start: end]
+        dpids = list(flows_acc.keys()) if isinstance(dpid, str) else dpid
+        if counter == 0:
+            counter = maximun
+            logger_fun(f"{message}{dpids}{total_length}, "
+                       f" flows({counter}): {flows_acc}")
+            total_length = ""
+            flows_acc = {}
+        start += len(flows[start:end])
+        end += counter
+    return counter, flows_acc, total_length
+
+
+def flows_to_log(
+    logger_fun: Callable,
+    message: str,
+    dpid: list,
+    content_dict: dict[str, dict],
+    by_switch=False,
+):
+    """New flog to log"""
     maximun = 200
-    flows_n = len(flow_dict["flows"])
-    i, j = 0, maximun
-    while flow_dict["flows"][i:j]:
-        logger_fun(
-            f"{message}{length_msg} flows[{i}, {(j if j < flows_n else flows_n)}]:"
-            f" {flow_dict['flows'][i:j]}"
+    flows_acc = {}
+    counter = maximun
+
+    if not by_switch:
+        flows = content_dict["flows"]
+        total_length = f", total_length: {len(flows)}"
+        counter, flows_acc, total_length = _flows_to_log(
+            logger_fun, flows, dpid, maximun, flows_acc, maximun, message, total_length
         )
-        i, j = j, j + maximun
-        length_msg = ""
+
+    else:
+        total_length = 0
+        for dpid in content_dict:
+            total_length += len(content_dict[dpid]["flows"])
```

Comment:
It's interesting the aggregated total_length, but what's usually valuable is knowing the total per switch just so you know if you're not overloading the switch and knowing upfront how many is sent, the aggregated total might be valuable but I don't know.

Reply:
Trying to log everything within the limits of 200 flows per log complicated the filtration of these. If it is better to know the number of flows per switch, maybe it would be better to log by switch and still apply the limitation of 200 flows in the log. This way we can keep the old method which is also more easy to understand. wdyt

Reply:
Method no longer necessary, removed. Logging is now by switch.